security/op-node-op-signer-integration.md

```diff
@@ -0,0 +1,145 @@
+
+## Usage of op-signer
+Previously, the only usage of op-signer was for the transaction signing of batcher and proposer. This uses the sender address of the tx and the secret key.
+
+However, for block payload signing, the sender address is not needed - we only need to sign the signingHash with the private key from the KMS. Therefore, we could make `p2p.signer.address` flag optional because it's required for the op-signer to be integrated with services like op-node (see `op-service/signer/cli.go`)
+
+After generating the appropriate key in the KMS, you have to add additional field to the config.yaml of op-signer as follows: 
+```yaml
+auth:
+  - name: op-node.primary.mainnet.prod.oplabs.cloud
+    key: projects/oplabs-prod-mainnet/locations/eur6/keyRings/testop-signer/cryptoKeys/op-node/cryptoKeyVersions/1
+// ... other existing keys
+  - name: sequencer-batcher.primary.mainnet.prod.oplabs.cloud
+    key: projects/oplabs-prod-mainnet/locations/eur6/keyRings/op-signer/cryptoKeys/batcher/cryptoKeyVersions/1
+```
+for each signing request, op-signer will look at the requesting server's domain, and look at this yaml file for a matching config. Then, it will look for the gcp kms key that matches the `key` name.
+
+
+## Security
+
+In such service where op-node and op-signer are located in a separate system and need to communicate, we need secure channel, where both parties can trust each other. For this, we use mutual TLS (mTLS). 
+
+With mTLS, we can prove that both the client and the server are verified with their own respective certificate. This allows op-node to trust the op-signer, and op-signer to sign only the requests from valid op-node sequencers. 
+
+Thankfully, op-signer already has mTLS support, as integrated in op-batcher and op-proposer. 
+
+Here, users can specify
+
+- tls.ca
+- tls.cert
+- tls.key
+
+and the mTLS is handled in op-service/tls. We have to set up the correct CA and so that both the op-signer and op-node can verify the certificates from the TLS handshake. 
+
+There has already been discussions about integrating op-signer with op-node, such as
+
+- https://github.com/ethereum-optimism/optimism/pull/4357#discussion_r1045342439
+- https://github.com/ethereum-optimism/optimism/pull/4669#discussion_r1067409473
+
+# Alternatives Considered
+- When we send the Block Payload from op-node to op-signer, we could send the full block payload arguments like the following: 
+```go
+type BlockPayloadArgs struct {
+	domain [32]byte `json:"domain"`
+	chainId *big.Int `json:"chainId"`
+	payloadHash [32]byte `json:"payloadHash"`
+}
+```
+This means we are sending more data to the op-signer, but may be more secure since the op-signer controls all of the logic for signing the payload. 
+
+# Risks & Uncertainties
```

Comment:
Another potential risk to keep in mind is ensuring we enforce the invariant that only the active sequencer is able to successfully sign blocks without odd race conditions (e.g., during sequencer leadership transfer).

To prevent such issues, does the design require any further integration / communication with op-conductor, or will this be guaranteed with the existing op-node implementation?